dtsync/syncer.go

```diff
@@ -60,12 +61,19 @@ func (s *Syncer) Sync(ctx context.Context, nextCid cid.Cid, sel ipld.Node) error
 		if err, ok := err.(rateLimitErr); ok {
 			log.Infow("hit rate limit. Waiting and will retry later", "cid", nextCid, "source_peer", s.peerID)
 
-			// Wait until we've fully refilled our rate limit bucket since this is a
-			// relatively heavy operation (essentially restarting the sync).
-			waitErr := s.rateLimiter.WaitN(ctx, s.rateLimiter.Burst())
-			if waitErr != nil {
-				return err
+			// Wait until the rate limit bucket is fully refilled since this is
+			// a relatively heavy operation (essentially restarting the sync).
+			// Note, cannot use s.rateLimiter.WaitN here because that waits,
+			// but also consumes n tokens.
+			waitTime := time.Second * time.Duration(s.rateLimiter.Burst()) / time.Duration(s.rateLimiter.Limit())
```

Comment:
what length of time is this going to be in reality?
while heavy, i wouldn't really imagine waiting more than about 1 second per resumption

Reply:
Only a second or two to restart sync depending on indexer load.  Well under a second in testing.  If the burst size if 500, and the rate is 100 per second, then the delay will be 5 seconds.